packages/FormElement/src/FormElement.js

```diff
@@ -71,85 +71,91 @@ function FormElement(props) {
         ariaDescriptionId,
       });
     }
 
     return null;
   }
 
+  function renderInstructions() {
+    if (extractedChildren["FormElement.Instructions"]) {
+      return React.cloneElement(extractedChildren["FormElement.Instructions"], {
+        ariaInstructionsId,
+      });
+    }
+  }
+
+  const nativeChildTypes = ["input", "textarea", "select"];
+
+  const renderFormElementChild = (child, index) => (
+    <div key={`child-key_${index}`} css={formElementChildStyle}>
+      {child}
+    </div>
+  );
+
+  const ariaDescribedByIds = `${ariaErrorId} ${ariaInstructionsId} ${ariaDescriptionId}`;
+
   return (
     <div css={formElementStyles} isInline={isInline} size={size} isDisabled={isDisabled} {...moreProps}>
       <Label
         hasOptionalLabel={hasOptionalLabel}
         hasRequiredLabel={hasRequiredLabel}
-        help={extratedChildren["FormElement.Help"]}
+        help={extractedChildren["FormElement.Help"]}
         id={inputId}
         isInline={isInline}
         isVisuallyHidden={isLabelVisuallyHidden}
         label={label}
       />
-
-      {extratedChildren.children.map(child => {
-        if (React.isValidElement(child)) {
-          const extendedProps = isString(child.type)
-            ? {
-                "aria-describedby": ariaDescriptionId,
-                id: inputId,
-                disabled: isDisabled,
-                readOnly: isReadOnly,
-              }
-            : {
-                "aria-describedby": ariaDescriptionId,
-                hasError,
-                id: inputId,
-                isDisabled,
-                isReadOnly,
-                size,
-              };
-          const clonedChild = React.cloneElement(child, extendedProps);
-
-          return (
-            <React.Fragment key={child.key}>
-              {isInline ? (
-                <div css={inlineContainerStyles}>
-                  {clonedChild}
-                  {renderFooter()}
-                </div>
-              ) : (
-                <React.Fragment>
-                  {clonedChild}
-                  {renderFooter()}
-                </React.Fragment>
-              )}
-            </React.Fragment>
-          );
-        }
-
-        return child;
-      })}
+      <div css={isInline ? inlineContainerStyles : null}>
+        {renderInstructions()}
+        {extractedChildren.children.map((child, index) => {
+          if (React.isValidElement(child)) {
+            const extendedProps = nativeChildTypes.includes(child.type)
+              ? {
+                  disabled: isDisabled,
+                  readOnly: isReadOnly,
+                  "aria-describedby": ariaDescribedByIds,
+                  id: inputId,
+                }
+              : {
+                  hasError,
+                  isDisabled,
+                  isReadOnly,
+                  size,
+                  "aria-describedby": ariaDescribedByIds,
+                  id: child.type.displayName === "Input" ? inputId : null,
```

Comment:
I think we should pass the `id` along to any React component, not just an `<Input>`.   I believe the `<DatePicker>` depends on this to associate the label to it, for instance.

Reply:
hmm, ok yeah i was trying to restrict it, ie in the case of checkbox component, but we can deal with that in a future pr